// @exponent/samples/ExponentConfigView.js
import React from 'react';
import {
  Image,
  Linking,
  Platform,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import { Constants } from 'exponent';

const SectionHeader = ({title}) => {
  return (
    <View style={styles.sectionHeaderContainer}>
	 <View style={styles.option}>
      <Image
            source={require('./assets/images/PS.png')}
			fadeDuration={0}
            style={{width: 20, height: 20}}
      />
	  <Text style={styles.sectionHeaderText}>
		  
        {title}
      </Text>
	  </View>
    </View>
  );
};

const SectionContent = (props) => {
  return (
    <View style={styles.sectionContentContainer}>
      {props.children}
    </View>
  );
};

export default class ExponentConfigView extends React.Component {

  render() {
    return (
      <View>
        <SectionHeader title=" Student & Parent Link" />
        <SectionContent>
			<TouchableOpacity onPress={this._handleStudentParentPress} style={styles.powerSchoolLink}>
              <Text style={styles.powerSchoolLinkText}>
                Student & Parent PowerSchool Link
              </Text>
            </TouchableOpacity>
        </SectionContent>

        <SectionHeader title=" Teacher Link" />
        <SectionContent>
           <TouchableOpacity onPress={this._handleTeacherPress} style={styles.powerSchoolLink}>
              <Text style={styles.powerSchoolLinkText}>
                Teacher PowerSchool Link
              </Text>
            </TouchableOpacity>
        </SectionContent>

        <SectionHeader title=" Administrator Link" />
        <SectionContent>
          <TouchableOpacity onPress={this._handleAdministratorPress} style={styles.powerSchoolLink}>
              <Text style={styles.powerSchoolLinkText}>
                Administrator PowerSchool Link
              </Text>
            </TouchableOpacity>
        </SectionContent>
      </View>
    );
  }
  
  _handleStudentParentPress = () => {
    Linking.openURL('https://ps.helenaschools.org/public/home.html');
  }
  _handleTeacherPress = () => {
    Linking.openURL('https://ps.helenaschools.org/teachers/pw.html');
  } 
   _handleAdministratorPress = () => {
    Linking.openURL('https://ps.helenaschools.org/administrators/pw.html');
  }
}

const styles = StyleSheet.create({
   sectionHeaderContainer: {
    backgroundColor: '#fbfbfb',
    paddingVertical: 8,
    paddingHorizontal: 15,
    borderWidth: StyleSheet.hairlineWidth,
    borderColor: '#ededed',
  },
  sectionHeaderText: {
    fontSize: 18,
	textAlignVertical: 'bottom',
  },
  sectionContentContainer: {
    paddingTop: 8,
    paddingBottom: 12,
    paddingHorizontal: 15,
  },
  sectionContentText: {
    color: '#999999',
    fontSize: 14,
  },
  powerSchoolContainer: {
    marginTop: 5,
    alignItems: 'center',
  },
  powerSchoolLink: {
    paddingVertical: 5,
  },
  powerSchoolLinkText: {
    fontSize: 16,
    color: '#2e78b7',
  },
   option: {
    flexDirection: 'row',
    backgroundColor: 'rgba(0,0,0,0.02)',
    paddingHorizontal: 15,
    paddingVertical: 15,
  },
});
