// exponent/src/Exponent.js
/**
 * @providesModule Exponent
*  @flow
 */

import {
  NativeModules,
} from 'react-native';

export const Amplitude = NativeModules.ExponentAmplitude;
export const Crypto = NativeModules.ExponentCrypto;
export const Fabric = NativeModules.ExponentFabric;
export const Facebook = NativeModules.ExponentFacebook;
export const FileSystem = NativeModules.ExponentFileSystem;
export const ImageCropper = NativeModules.ExponentImageCropper;
export const ImagePicker = NativeModules.ExponentImagePicker;
export const Notifications = NativeModules.ExponentNotifications;
export const Segment = NativeModules.ExponentSegment;
export const Util = NativeModules.ExponentUtil;

import './Logs';

export * as Logs from './Logs';

export { default as registerRootComponent } from './registerRootComponent';
export { default as takeSnapshotAsync } from './takeSnapshotAsync';
export { default as Asset } from './Asset';
export { default as apisAreAvailable } from './apisAreAvailable';
export * as Constants from './Constants';
export * as Contacts from './Contacts';
export * as Font from './Font';
export * as Location from './Location';
export * as Permissions from './Permissions';

export * as Components from './Components';
